pkg/exporter/process.go

```diff
@@ -164,26 +165,22 @@ func (ep *ExportingProcess) NewTemplateID() uint16 {
 	return ep.templateID
 }
 
-func (ep *ExportingProcess) sendMsg(set entities.Set) (int, error) {
+func (ep *ExportingProcess) sendMsg(msg *entities.Message, set entities.Set) (int, error) {
```

Comment:
`sendMsg` and `sendSet` is kinda dependent/overlapping on each other. In `sendSet` there is creation of new message but not filling it completely. `sendMsg` will take the incomplete message and fill it. Will it be better to only checking the set in `sendSet` and call function like `createMsgAndSend(set)`. in `createMsgAndSend(set)` it will create new message, add set(s) into it and send it. What do you think?

Reply:
Good idea. Initially, I thought about that but felt the logic in`SendSet` will become very lean and almost like sanity check of the set. I agree that it is a cleaner way to do that and can be extended easily for multiple sets. Modified it.
In addition, I optimized the function a bit more to avoid writing the set buffer to message buffer. I append the slices and send it on the exporter connection.